pkg/provisioner/ironic/testserver/ironic.go

```diff
@@ -230,3 +243,70 @@ func (m *IronicMock) WithNodeValidate(nodeUUID string) *IronicMock {
 	m.ResponseWithCode("/v1/nodes/"+nodeUUID+"/validate", "{}", http.StatusOK)
 	return m
 }
+
+// ClearIronicDatabase simulates the loss of the Ironic database
+//
+// Database clearing is simulated by setting all registered non-POST handlers to
+// return a 404 error.  Both regular handlers, and default handlers are
+// affected.
+func (m *IronicMock) ClearIronicDatabase() {
+	if m.databaseCleared {
+		return
+	}
+
+	m.t.Log("clearing ironic database")
+
+	// First, set all default handlers to reply with 404
+	newDefaultResponses := []defaultResponse{}
+
+	for _, resp := range m.defaultResponses {
+
+		// POST requests should still succeed when the database is empty
+		if resp.method == "POST" {
+			newDefaultResponses = append(newDefaultResponses, resp)
+			continue
+		}
+
+		newDefaultResponse := defaultResponse{
+			method: resp.method,
+			re:     resp.re,
+			response: response{
+				code:    http.StatusNotFound,
+				payload: "",
+			},
+		}
+		newDefaultResponses = append(newDefaultResponses, newDefaultResponse)
+	}
+
+	m.defaultResponses = newDefaultResponses
+
+	// Second, set named handlers to reply with 404
+	newResponsesByMethod := make(map[string]map[string]response)
+
+	for url, methodMap := range m.responsesByMethod {
+		newResponsesByMethod[url] = map[string]response{}
+
+		for method, resp := range methodMap {
+			// POST requests should still succeed when the database is empty
+			if method == "POST" {
+				newResponsesByMethod[url][method] = resp
+				continue
+			}
+
+			newResponsesByMethod[url][method] = response{
+				code:    http.StatusNotFound,
+				payload: "",
+			}
+		}
+	}
```

Comment:
Also this one could be simplified, in this case by simply reassigning a modified resp